src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -1264,39 +1276,38 @@ def visit_FunCall(self, node: gtir.FunCall) -> IteratorExpr | DataExpr:
         elif cpm.is_applied_shift(node):
             return self._visit_shift(node)
 
+        elif isinstance(node.fun, gtir.Lambda):
+            raise AssertionError("Lambda node should be visited with 'apply()' method.")
```

Comment:
I am actually not sure if you should do that, raise an `AssertionError`.
One reason is that the `-O` flag does not work, which is supposed to disable assertions.
However, this is more an input error, so I would either raise a `ValueError` or a `NotImplementedError`.